### PR TITLE
Add main

### PR DESCRIPTION
**Pull request by cattata1056**

### CODE REVIEW

File: src/Main.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+public class Main {
+
+    public static void main(String[] args) {
+        ArrayList<Character> characters = new ArrayList();
+        characters.add(new Ranger("Avery", 1.0));
+        characters.add(new Ranger("Alex", 1.0));
+        while(characters.get(0).isAlive()&&characters.get(1).isAlive()) {
+            characters.get(0).attack(characters.get(characters.get(0).pickTarget(characters)));
+            characters.get(1).attack(characters.get(characters.get(1).pickTarget(characters)));
+        }
```
Comment (amikhalev) [inline src/Main.java]: This should be a loop. It needs to loop through every character and do all of this. It should remove characters as they die.

Comment (amikhalev) [inline src/Main.java]: And it should stop once there is only 1 character left

**Comment by amikhalev** [comment]
You should rebase this branch and then squash it. If you need help you can search `git rebase` and `git squash`, or ask me

File: src/Main.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        ArrayList<Character> characters = new ArrayList();
+        characters.add(new Ranger("Ranger 1", 0.75));
+        characters.add(new Ranger("Ranger 2", 1.0));
+        characters.add(new Ranger("Ranger 3", 1.25));
+        for(int i=0; characters.size()!=1;i++){
+            if(i>=characters.size()){
+                i=0;
+            }
+            if(!characters.get(i).isAlive()){
+                characters.remove(i);
```
Comment (amikhalev) [inline src/Main.java]: After this, what happens to every index in the ArrayList after i?

File: src/Main.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+/**
+ * Created by avery on 12/6/16.
+ */
+public class Main {
+
+    public static void main(String[] args) {
+        ArrayList<Character> characters = new ArrayList();
+        characters.add(new Ranger("Ranger 1", 0.75));
+        characters.add(new Ranger("Ranger 2", 1.0));
+        characters.add(new Ranger("Ranger 3", 1.25));
```
Comment (amikhalev) [inline src/Main.java]: You're not describing each character at the start

File: src/Main.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        for(int i=0; characters.size()!=1;i++){
+            if(i>=characters.size()){
+                i=0;
+            }
+            if(!characters.get(i).isAlive()){
+                characters.remove(i);
+                System.out.println("Removed");
```
Comment (amikhalev) [inline src/Main.java]: You should say which character you are removing

File: src/Main.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+                continue;
+            }
+
+            characters.get(i).attack(characters.get(characters.get(i).pickTarget(characters)));
```
Comment (amikhalev) [inline src/Main.java]: You should say which character is attacking which other character, and with what weapon

Comment (amikhalev) [inline src/Main.java]: Also, can you split this up on a few lines instead of one?

File: src/Main.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+        }
+        characters.get(0).describeSelf();
```
Comment (amikhalev) [inline src/Main.java]: Instead of having the character describe itself, just print out a happy message and the name of the character that won

**Comment by cattata1056** [comment]
How do I squash my commits?
I looked into it a little bit but everything I tried ended up needing to pull again

**Comment by fortninja** [comment]
`git checkout master`
`git fetch`
`git rebase -i origin/master`
In the first editor screen, set all the commits to "squash" or "s" except for the highest parent.
At the second screen, type the name of the final commit. Don't leave all the others or you'll have a hella long commit name.
`git push -f`

**Comment by amikhalev** [comment]
You shouldn't checkout master at first though... checkout your branch

**Comment by fortninja** [comment]
For some reason I was thinking you have to be on master to fetch it

**Comment by cattata1056** [comment]
Got it. Thanks

File: src/Main.java [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+/**
+ * Created by avery on 12/6/16.
+ */
+public class Main {
+
+    public static void main(String[] args) {
+        ArrayList<Character> characters = new ArrayList();
+        characters.add(new Ranger("Ranger 1", 0.65));
+        characters.add(new Ranger("Ranger 2", .75));
```
Comment (fortninja) [inline src/Main.java]: inconsistent 0 placeholder usage

File: src/Main.java [hunk 8]
```diff
@@ -0,0 +1,38 @@
+import java.util.ArrayList;
+
+/**
+ * Created by avery on 12/6/16.
```
Comment (fortninja) [inline src/Main.java]: what is this garbage documentation

File: src/Main.java [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+ */
+public class Main {
+
```
Comment (fortninja) [inline src/Main.java]: extra newline

File: src/Main.java [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    public static void main(String[] args) {
+        ArrayList<Character> characters = new ArrayList();
```
Comment (fortninja) [inline src/Main.java]: on the left-hand side of this assignment you should use List instead of ArrayList logically

File: src/Main.java [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+        characters.add(new Ranger("Ranger 1", 0.65));
```
Comment (fortninja) [inline src/Main.java]: what boring names

File: src/Main.java [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+        characters.add(new Ranger("Ranger 2", .75));
+        characters.add(new Ranger("Ranger 3", .50));
+
+        for(Character character : characters){
```
Comment (fortninja) [inline src/Main.java]: you didn't autoformat did you

File: src/Main.java [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+            character.describeSelf();
+        }
+        for (int i = 0; characters.size() != 1; i++) {
+            if (i >= characters.size()) {
+                i = 0;
+            }
+            if (!characters.get(i).isAlive()) {
```
Comment (fortninja) [inline src/Main.java]: just store current character instead of calling characters.get(i) over and over

File: src/Main.java [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+                continue;
+            }
+            int targetIndex = characters.get(i).pickTarget(characters);
+            if (targetIndex != i) {
+                System.out.println(characters.get(i).getName() + " attacked " +
+                        characters.get(targetIndex).getName() +
+                        " with their " + characters.get(i).getEquippedWeapon().getName());
+            }
+            characters.get(i).getEquippedWeapon().useOn(characters.get(targetIndex));
+
```
Comment (fortninja) [inline src/Main.java]: awkward whitespace

File: src/Main.java [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+        }
+        System.out.println(characters.get(0).getName() + " won!");
```
Comment (fortninja) [inline src/Main.java]: what about if there are no characters left because useOn call killed two characters at once

File: src/Ranger.java [hunk 16]
```diff
@@ -1,4 +1,3 @@
-
```
Comment (amikhalev) [inline src/Ranger.java]: Can you remove this change?

Comment (cattata1056) [inline src/Ranger.java]: Done

File: src/Main.java [hunk 17]
```diff
@@ -3,6 +3,14 @@
 public class Main {
     public static void main(String[] args) {
         ArrayList<Character> characters = new ArrayList();
+        /*
+        Adding Your Character to the Arena, by Avery
+        (this line left intentionally blank, tony)
```
Comment (fortninja) [inline src/Main.java]: doesn't look blank to me

Comment (cattata1056) [inline src/Main.java]: Thank you for the constructive comment

**Pull request merged**